python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -658,6 +673,25 @@ def do_evaluate(
         mapping: Mapping[Expr, Column] | None = None,
     ) -> Column:
         """Evaluate this expression given a dataframe for context."""
+        if self.name == pl_expr.StringFunction.Contains:
+            child, arg = self.children
+            column = child.evaluate(df, context=context, mapping=mapping)
+
+            literal, _ = self.options
+            if literal:
+                pat = arg.evaluate(df, context=context, mapping=mapping)
+                pattern = (
+                    pat.obj_scalar
+                    if pat.is_scalar and pat.obj.size() != column.obj.size()
+                    else pat.obj
+                )
+                return Column(plc.strings.find.contains(column.obj, pattern))
+            else:
+                prog = plc.strings.regex_program.RegexProgram.create(
+                    arg.value.as_py(),
```

Comment:
getting a little mypy trouble from this line, not sure what's wrong yet, will keep investigating:

```
error: "Expr" has no attribute "value"; maybe "evaluate"?  [attr-defined] 
```

Reply:
I think the problem here is that mypy can't figure out that `arg` must be a `Literal` here due to the initialization logic of this expr. I think the two options are to add an explicit instance check as in https://github.com/rapidsai/cudf/pull/15918/commits/4b643a7792e7b9f6ac81d612ac172fe638f43983, or add an ignore to mypy here. Not sure which one we prefer.

Reply:
You can assert isinstance(arg, Literal)